src/crypto/ripemd160.cpp

```diff
@@ -277,11 +277,22 @@ void CRIPEMD160::Finalize(unsigned char hash[OUTPUT_SIZE])
     WriteLE64(sizedesc, bytes << 3);
     Write(pad, 1 + ((119 - (bytes % 64)) % 64));
     Write(sizedesc, 8);
+    Midstate(hash, NULL, NULL);
+}
+
+void CRIPEMD160::Midstate(unsigned char hash[OUTPUT_SIZE], unsigned char* buffer, size_t* length)
+{
     WriteLE32(hash, s[0]);
     WriteLE32(hash + 4, s[1]);
     WriteLE32(hash + 8, s[2]);
     WriteLE32(hash + 12, s[3]);
     WriteLE32(hash + 16, s[4]);
+    if (length) {
+        *length = bytes << 3;
```

Comment:
Because that's the FIPS-180 standard, is the short answer. SHA-256 includes the number of bits hashed in the final compression round as an unsigned 64-bit integer, meaning that the bit counter wraps at 2^61 bytes (2^125 for SHA-512).

SHA-1 and RIPEMD-160 use the same bit-oriented padding function as SHA-256.